src/main/java/yevhen/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,40 @@
+package yevhen.bookstore.repository.impl;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+import yevhen.bookstore.model.Book;
+import yevhen.bookstore.repository.BookRepository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public Book save(Book book) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```

Comment:
Don't use try with resources here because session will be closed if something happened and rollback doesn't work well